source/lattice-construction.md

```diff
@@ -0,0 +1,158 @@
+what is designated as an existing branch is not possible since the branch being forked to 
+does not yet exist since
+the `Fork` element that instantiates this branch has not been processed yet. In this case,
+the first `Fork` element must be temporarily ignored until the branch is instantiated.
+
+%---------------------------------------------------------------------------------------------------
+(s:forking)=
+## Forking
+
+A `Fork` element marks a point in a lattice branch where another branch is attached to. 
+A common example is using a `Fork` element to connect from a storage ring to 
+an extraction line or an X-ray beam line. Similarly, a `Fork` can be used to connect 
+from the end of an injection line to someplace in a ring. 
+An example of how forking can be used to construct complex machine layouts
+is shown in figure {numref}`f:fork`. 
+
+```{figure} figures/fork.svg
+:width: 80%
+:name: f:fork
+
+Example of how `Fork` and `Patch` elements can be used to describe even complicated machine geometries. 
+Shown is a section of the 8-pass (4 passes with increasing energy and 4 passes with decreasing energy) 
+Cornell/Brookhaven CBETA ERL/FFAG machine. `Fork` elements are used to connect the 
+injection line to the ERL and to connect the ERL to a diagnostic line. The geometry of the
+switchyard, used to correct the timings of the beams with differing energies from the FFAG arc, 
+is done using `Patch` elements.
+```
+
+The `branch` containing a forking element is called the
+"`from-branch`". The `branch` that the forking element points to is called the
+"`to-branch`". It is possible for these two branches to be one and the same.
+The element in the to-branch that the `Fork` connects to is called the "`to-element`".
+
+`Fork` elements are uni-directional. That is, particles can travel from a `Fork` element
+in a `from-branch` to the `to-branch` of the `Fork` but travel cannot happen in
+the reverse direction. To get a bi-directional link between branches, use two `Forks` with
+the `to-element` of both `Forks` being the other `Fork`.
+
+To avoid ambiguities, the types of `to-elements` are restricted to be one of:
+- Marker
+- Beginning
+- FloorPosition
+- Fork
+Notice that these types of elements have zero length and unit transfer maps.
+
+A `Fork` element has zero length and must always have a [`ForkP`](#fork.group) parameter group.
+The components of the `ForkP` group are:
+```{code} yaml
+  to_BeamLine           # String: Name of BeamLine to fork to.
+  to_Branch             # String: Name of Branch to fork to.
+  to_element            # Optional String: Element to fork to. Default is Beginning.
+  direction             # Optional Switch: Longitudinal Direction of travel of injected beam.
+  Branch_name           # Optional String: Name to give created Branch.
+  propagate_reference   # Optional Boolian: Propagate reference species and energy? Default is False.
```

Comment:
Replace '?' with '.'?

Reply:
The "?" is meant to indicate that the propagation only happens if `propagate_reference` is set to `True`.